internal/audit/report.go

```diff
@@ -0,0 +1,146 @@
+package audit
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gopasspw/gopass/internal/hashsum"
+	"github.com/gopasspw/gopass/internal/set"
+)
+
+type SecretReport struct {
+	Name     string
+	Errors   []error
+	Warnings []string
+	Age      time.Duration
+}
+
+func (s SecretReport) Record() []string {
+	return []string{
+		s.Name,
+		s.Age.String(),
+		strings.Join(errors(s.Errors), ";"),
+		strings.Join(s.Warnings, ";"),
+	}
+}
+
+func errors(e []error) []string {
+	s := make([]string, 0, len(e))
+	for _, es := range e {
+		s = append(s, es.Error())
+	}
+
+	return s
+}
+
+type Report struct {
+	Secrets  map[string]SecretReport
+	Template string
+}
+
+type ReportBuilder struct {
+	sync.Mutex
+
+	secrets map[string]SecretReport
+	// SHA512(password) -> secret names
+	duplicates map[string]set.Set[string]
+
+	// HIBP
+	// SHA1(password) -> secret names
+	sha1sums map[string]set.Set[string]
```

Comment:
I commented elsewhere already, but re-iterating that I don't see how we could get two secrets with the same path so I don't see why we need `set.Set` for these?

Reply:
This map is from sha1sum to secret name(s).

If we have multiple secrets that contain the same secret (very likely scenario) we need to handle that.

Reply:
But then we really need to map from 1 string to a slice of strings. `map[string][]string` should be enough, no?

Reply:
Yes, that would work, too.